tensorboard/uploader/uploader_test.py

```diff
@@ -355,6 +356,65 @@ def test_upload_skip_large_blob(self):
         self.assertEqual(0, mock_rate_limiter.tick.call_count)
         self.assertEqual(1, mock_blob_rate_limiter.tick.call_count)
 
+    def test_filter_graphs(self):
```

Comment:
Thanks for cleaning up and simplifying these tests!  However, I am a little skeptical of this design because the tests are no longer independent.  Can you factor out some common setup but keep the tests separate (while retaining most of the simplicity)?

Also, why did you remove the explicit tests for `RuntimeWarning` vs. `DecodeError` that @caisq had just requested in #3508?

Finally FYI #3509 is outstanding.  NP, I'll merge it, assuming you submit first.  The behavior of the empty graph case may actually change, depending on how you address the 'corrupt graph' case above.

Reply:
OK for now for expediency.  For the record, my concern is that the setup code is not generic; it includes setup specific to the tests (e.g., the values and order of the bytes examples).  If I wanted to add another subtest--perhaps the one from #3509--I'd have to change the setup code here.  I can see this both ways (i.e., I could also add more tests that use the same examples, without changing the setup).  The fact that currently parts of the setup correspond 1:1 with specific tests makes it look more entangled than it necessarily is.